nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/util/SnippetUtils.java

```diff
@@ -235,41 +236,24 @@ public FlowSnippetDTO populateFlowSnippet(final Snippet snippet, final boolean r
 
         // add any process groups
         final ProcessGroupDTO highestProcessGroupDTO = dtoFactory.createProcessGroupDto(processGroup, recurse);
-        final Set<ProcessGroupDTO> processGroups = highestProcessGroupDTO.getContents().getProcessGroups();
         fillContentsByGroupMap(highestProcessGroupDTO, contentsByGroup);
```

Comment:
This would probably result in an error if a selected group refers to a controller service that is in an unselected group because that foreign controller service wouldn't inherit the id of the highest process group but would retain that of the unselected group.